crates/sui-framework/packages/sui-system/sources/sui_system.move

```diff
@@ -534,79 +523,82 @@ module sui_system::sui_system {
 
     /// Return the current epoch number. Useful for applications that need a coarse-grained concept of time,
     /// since epochs are ever-increasing and epoch changes are intended to happen every 24 hours.
-    public fun epoch(wrapper: &SuiSystemState): u64 {
+    public fun epoch(wrapper: &mut SuiSystemState): u64 {
```

Comment:
It feels off to have a getter take in a mutable reference but I guess with this approach we don't have a way around it. Also this is a public fun so we won't be able to change its signature in any framework upgrade later. Let's make sure we want to do this.

Reply:
Actually I don't think these getters are used outside of tests right now. Do we want to make them test only and only upgrade to non test only if we need to?

Reply:
My general question/thought is how much interface we want to provide for dapp developers who wants to interact with system state in their contracts